### PR TITLE
MAINT: remove deprecation warning when acq_optimizer='auto'

### PR DESCRIPTION
**Pull request by mirca**
Based on the chat in #354, I'm removing the deprecation warning when ``acq_optimizer='auto'``.

I think after #354, it would make sense to let ``acq_optimizer='auto'`` as default... What do you think?

Does this need an entry in the change log?

### CODE REVIEW

**Comment by MechCoder** [comment]
Yes, can you add an entry in the change-log.

Also, just FYI the "auto" option was there to configure the `acq_optimizer` option on the basis of the space configuration which was removed by me over here (https://github.com/scikit-optimize/scikit-optimize/pull/285/)

**Comment by MechCoder** [comment]
It should be set to sampling here as well (https://github.com/scikit-optimize/scikit-optimize/blob/master/skopt/optimizer/forest.py#L12).  :/

**Comment by betatim** [comment]
We removed `auto` as a valid parameter value and are now adding it back with a different meaning. How confused are users going to be?

---

> It should be set to sampling here as well (https://github.com/scikit-optimize/scikit-optimize/blob/master/skopt/optimizer/forest.py#L12).  :/

can we remove it all together from `forest_optimize` (and the other tree based optimize calls if they list it). For now `sampling` is the correct and only choice so we don't need a parameter :)

**Comment by MechCoder** [comment]
There are two things.

1. The `acq_optimizer` argument as supplied to the Optimizer directly. Since the Optimizer class is a new class the "auto" argument for the `acq_optimizer` did not exist before.
2.  The deprecation warnings are for the `acq_optimizer` as supplied through the *_minimize functions depending upon the Space configuration.

We can move the warning out of the `Optimizer` class and into the `base_minimize` function.

**Comment by mirca** [comment]
@MechCoder @betatim Just to be crystal clear, should add a deprecation warning for ``acq_optimizer=auto`` in ``base_minimize``?

File: skopt/optimizer/forest.py [hunk 1]
```diff
@@ -7,12 +7,10 @@
 from ..learning import RandomForestRegressor
 
 
-def forest_minimize(func, dimensions, base_estimator="ET",
-                    n_calls=100, n_random_starts=10,
-                    acq_func="EI", acq_optimizer="auto",
```
Comment (MechCoder) [inline skopt/optimizer/forest.py]: This will break user-code. Let's keep it for now and hopefully we will add something better later on?

**Comment by MechCoder** [comment]
Yes, that is what I had meant. WDYT?

File: skopt/optimizer/base.py [hunk 2]
```diff
@@ -162,6 +162,10 @@ def base_minimize(func, dimensions, base_estimator,
         "n_jobs": n_jobs}
     acq_func_kwargs = {"xi": xi, "kappa": kappa}
 
+    if acq_optimizer == "auto":
+        warnings.warn("The 'auto' option for the acq_optimizer will be "
+                      "removed in 0.4.")
```
Comment (betatim) [inline skopt/optimizer/base.py]: Shouldn't we remove this? The next release will be v0.4 so we can't remove the option?

Comment (MechCoder) [inline skopt/optimizer/base.py]: Yes indeed we should remove it from `base_minimize`

File: skopt/optimizer/optimizer.py [hunk 3]
```diff
@@ -69,11 +69,15 @@ class Optimizer(object):
                 - After fitting the surrogate model with `(X_best, y_best)`,
                   the gains are updated such that $g_i -= \mu(X_i)$
 
-    * `acq_optimizer` [string, `"sampling"` or `"lbfgs"`, default=`"lbfgs"`]:
+    * `acq_optimizer` [string, `"sampling"`, `"lbfgs"`, or `"auto"`, default=`"lbfgs"`]:
```
Comment (MechCoder) [inline skopt/optimizer/optimizer.py]: you should remove if from here as well.

**Comment by MechCoder** [comment]
Merging as non-controversial.

**Comment by MechCoder** [comment]
Thanks!

**Comment by betatim** [comment]
🍻 thanks for this!

**Comment by mirca** [comment]
@betatim @MechCoder Awesome! x)

**Pull request merged**